crates/biome_js_analyze/src/analyzers/nursery/use_shorthand_assign.rs

```diff
@@ -0,0 +1,268 @@
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```js
+    /// var a = 1;
+    /// a += 1;
+    /// ```
+    ///
+    /// ```js
+    /// var a = 1;
+    /// a -= 1;
+    /// ```
+    ///
+    ///  ```js
+    /// var a = 1;
+    /// a *= 1;
+    /// ```
+    ///
+    ///  ```js
+    /// var a = 1;
+    /// a /= 1;
+    /// ```
+    ///
+    /// ```js
+    /// var a = 1;
+    /// a %= 1;
+    /// ```
+    ///
+    /// ```js
+    /// var a = 1;
+    /// a <<= 1;
+    /// ```
+    ///
+    /// ```js
+    /// var a = 1;
+    /// a >>= 1;
+    /// ```
+    ///
+    /// ```js
+    /// var a = 1;
+    /// a >>>= 1;
+    /// ```
+    ///
+    /// ```js
+    /// var a = 1;
+    /// a &= 1;
+    /// ```
+    ///
+    /// ```js
+    /// var a = 1;
+    /// a ^= 1;
+    /// ```
+    ///
+    /// ```js
+    /// var a = 1;
+    /// a |= 1;
+    /// ```
+    ///
+    /// ```js
+    /// var a = 1;
+    /// a **= 1;
+    /// ```
+    ///
+    pub(crate) UseShorthandAssign {
+        version: "next",
+        name: "useShorthandAssign",
+        recommended: false,
+    }
+}
+
+pub struct RuleState {
+    shorthand_operator: JsSyntaxKind,
+    right: AnyJsExpression,
+}
+
+impl Rule for UseShorthandAssign {
+    type Query = Ast<JsAssignmentExpression>;
+    type State = RuleState;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let operator = node.operator().ok()?;
+        let right = node.right().ok();
+        let left = node.left().ok();
+        let left_var_name = left?.text();
+
+        if !matches!(operator, JsAssignmentOperator::Assign) {
+            return None;
+        }
+
+        let binary_expression = match right {
+            Some(AnyJsExpression::JsBinaryExpression(binary_expression)) => Some(binary_expression),
+            Some(AnyJsExpression::JsParenthesizedExpression(param)) => Some(
+                JsBinaryExpression::cast_ref(param.expression().ok()?.syntax())?,
+            ),
+            _ => None,
+        }?;
+
+        let has_same_reference =
+            has_same_reference_in_expression(left_var_name, &binary_expression)?;
+        let operator = binary_expression.operator().ok()?;
+        let right = binary_expression.right().ok()?;
+        let shorhand = get_shorthand(&operator)?;
+
+        if has_same_reference {
+            Some(RuleState {
+                shorthand_operator: shorhand,
+                right,
+            })
+        } else {
+            None
+        }
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        let shorthand_operator = state.shorthand_operator.to_string()?;
+
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            node.range(),
+            markup! {
+                "Assignment (=) can be replaced with operator assignment "{shorthand_operator}""
```

Comment:
We should use `<Emphasis>` to highlight the `=` and the `shorthand_operator`:

https://github.com/biomejs/biome/blob/599665712c66c5e72198193ce926770c2b838507/crates/biome_js_analyze/src/analyzers/style/no_parameter_properties.rs#L57-L64